pkg/controller/v1beta1/inferenceservice/controller_test.go

```diff
@@ -397,6 +400,132 @@ var _ = Describe("v1beta1 inference service controller", func() {
 			Expect(updatedVirtualService.Annotations).To(Equal(annotations))
 			Expect(updatedVirtualService.Labels).To(Equal(labels))
 		})
+		It("Should create a knative service with initial-scale:0 and min-scale:0 annotations", func() {
```

Comment:
I think you are using `By` to separate test cases but `By` is not for that
`By` is used to explicitly describe the detailed steps of a test or the actions that occur at each step. It is typically used within the `It` block to specify the actions that should be performed. For example, actions such as submitting a form, making a specific API call, or creating an object can be described using `By`.

Instead of using `By`, why don't you use `Context/When/It`